src/sentry/api/invite_helper.py

```diff
@@ -1,33 +1,53 @@
 from __future__ import annotations
 
+import dataclasses
 from logging import Logger
-from typing import Any, Dict, Tuple
+from typing import Dict, Optional
 
 from django.utils.crypto import constant_time_compare
 from rest_framework.request import Request
 
 from sentry import audit_log, features
-from sentry.models import AuthIdentity, AuthProvider, OrganizationMember, User, UserEmail
-from sentry.signals import member_joined
+from sentry.models import AuthIdentity, AuthProvider, User, UserEmail
+from sentry.services.hybrid_cloud.organization import (
+    RpcOrganizationMember,
+    RpcUserInviteContext,
+    organization_service,
+)
 from sentry.utils import metrics
 from sentry.utils.audit import create_audit_entry
 
 
-def add_invite_details_to_session(request: Request, member_id: int, token: str) -> None:
+def add_invite_details_to_session(
+    request: Request, member_id: int, token: str, organization_id: int
+) -> None:
     """Add member ID and token to the request session"""
     request.session["invite_token"] = token
     request.session["invite_member_id"] = member_id
+    request.session["invite_organization_id"] = organization_id
 
 
 def remove_invite_details_from_session(request: Request) -> None:
     """Deletes invite details from the request session"""
     request.session.pop("invite_member_id", None)
     request.session.pop("invite_token", None)
+    request.session.pop("invite_organization_id", None)
 
 
-def get_invite_details(request: Request) -> Tuple[str, int]:
+@dataclasses.dataclass
+class InviteDetails:
+    invite_token: Optional[str]
+    invite_member_id: Optional[int]
+    invite_organization_id: Optional[int]
```

Comment:
Nice.

I think there might be an opportunity to fold the two functions above into this class, but we can save that for a later refactoring.